### ISSUE TITLE
Not Match tag right after insert

### ISSUE THREAD

**Issue by vahid-sohrabloo** [VictoriaMetrics/VictoriaMetrics#163]
**Describe the bug**
If Insert  a data and then immediately search with a tag filter will not show any data

**To Reproduce**
curl -d 'm,unit=1001 io1=10u 1566888197000000000' -X POST 'http://localhost:8428/write' && curl -G 'http://localhost:8428/api/v1/export?format=promapi' -d 'match={unit="1001"}'

**Version**
v1.26.0

**Comment by valyala** [comment]
This is expected behavior. The ingested data is buffered before going to the storage for performance reasons. The data appears after a few seconds, so insert something like `sleep 2` between the command that inserts data and the command that selects it.

What is your use case?

**Comment by vahid-sohrabloo** [comment]
After write and immediately search, If I query again event after 20 seconds there is no result.

**Comment by valyala** [comment]
I cannot reproduce the issue on v1.26.0. It successfully returns the inserted data point:

```
$ curl -d 'm,unit=1001 io1=10u 1566888197000000000' -X POST 'http://localhost:8428/write' 
$ curl -G 'http://localhost:8428/api/v1/export?format=promapi' -d 'match={unit="1001"}'
{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"__name__":"m_io1","unit":"1001"},"values":[[1566888197,"10"]]}]}}
```

**Comment by vahid-sohrabloo** [comment]
You should export data immediately after a write

```
 curl -d 'm,unit=1001 io1=10u 1566888197000000000' -X POST 'http://localhost:8428/write' && curl -G 'http://localhost:8428/api/v1/export?format=promapi' -d 'match={unit="1001"}'
```

**Comment by valyala** [comment]
Yes, this doesn't work because of buffering I described in the https://github.com/VictoriaMetrics/VictoriaMetrics/issues/163#issuecomment-525527641 . You need to wait for a few seconds before exporting the inserted data

**Comment by vahid-sohrabloo** [comment]
Is this expected behavior?
``
curl -d 'm,unit=1001 io1=10u 1566888197000000000' -X POST 'http://localhost:8428/write' && curl -G 'http://localhost:8428/api/v1/export?format=promapi' -d 'match={unit="1001"}' && sleep 10  &&  curl -G 'http://localhost:8428/api/v1/export?format=promapi' -d 'match={unit="1001"}'
``{"status":"success","data":{"resultType":"matrix","result":[]}}{"status":"success","data":{"resultType":"matrix","result":[]}}```

**Comment by valyala** [comment]
This is unexpected behavior. It will be fixed soon. Thanks for the details on the bug report, @vahid-sohrabloo !

**Comment by valyala** [comment]
The issue should be fixed in the commit e1d76ec1f3d92d1b8e7323d359cc05fc7030715f . @vahid-sohrabloo , could you build VictoriaMetrics from this commit according to [these instructions](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources) and verify whether the issue fixed there?

**Comment by vahid-sohrabloo** [comment]
It's fixed. Thanks
